### PR TITLE
bump version to 0.17.0

### PR DESCRIPTION
**Pull request by apoelstra**

### CODE REVIEW

**Comment by apoelstra** [comment]
It's in this release, the "BIP32 cleanup" link goes to it. Should I reword that?

Also @TheBlueMatt I'd like your ACK on this.

**Comment by stevenroose** [comment]
Ah was in mobile. Nah I guess that's fine.

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,4 +1,16 @@
...
+* [Replace `rust-crypto` with `bitcoin_hashes`; refactor hash types](https://github.com/rust-bitcoin/rust-bitcoin/pull/215)
+* [Remove `Address::p2pk`](https://github.com/rust-bitcoin/rust-bitcoin/pull/222/)
+* Remove misleading blanket `MerkleRoot` implementation; [it is now only defined for `Block`](https://github.com/rust-bitcoin/rust-bitcoin/pull/218)
+* [Add BIP157](https://github.com/rust-bitcoin/rust-bitcoin/pull/215) (client-side block filtering messages)
+* [Replace all key types](https://github.com/rust-bitcoin/rust-bitcoin/pull/183) to better match abstractions needed for PSBT
+* [Clean up BIP32](https://github.com/rust-bitcoin/rust-bitcoin/pull/233) in preparation for PSBT; [use new native key types rather than `secp256k1` ones](https://github.com/rust-bitcoin/rust-bitcoin/pull/238/)
+* Remove [apparently-used `Option` serialization](https://github.com/rust-bitcoin/rust-bitcoin/pull/236#event-2158116421) code
+* Finally merge [PSBT](https://github.com/rust-bitcoin/rust-bitcoin/pull/103) after nearly nine months
+
 # 0.16.0 - 2019-01-15
```
Comment (dr-orlovsky) [inline CHANGELOG.md]: Let me suggest adding to the list a recently merged PR for serialization for a network stream: https://github.com/rust-bitcoin/rust-bitcoin/pull/231

Comment (apoelstra) [reply CHANGELOG.md]: Sure.

Comment (apoelstra) [reply CHANGELOG.md]: Look good now?

**Pull request merged**